### PR TITLE
(FMT): allow #[macro_use] attr to be placed in line with extern crate item

### LINKED ISSUE #548: Allow #[macro_use] attr to be placed in line with extern crate item
subject

### PR DESCRIPTION
**Pull request by jajakobyly**
fixes #548

### CODE REVIEW

File: src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#[macro_use] extern crate foo;
```
Comment (matklad) [inline src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs]: So we always place `macro_use` inline? I think this is a bit too opinionated: some do this, some don't:

https://github.com/search?utf8=%E2%9C%93&q=%23%5Bmacro_use%5D&type=Code&ref=searchresults

It's better to either have an option here, or to be able to preserve existing formatting.

**Comment by matklad** [comment]
Nice test: https://github.com/cuviper/rust-libprobe/blob/c889fce367aabc78fcd7f347eea7fff569fb6268/examples/loop.rs

File: src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs (hunk 1 above)
Comment (jajakobyly) [inline src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs]: And what should be the default?

**Comment by matklad** [comment]
Also, was there `#[macro_use] mod foo;` as well?

**Comment by jajakobyly** [comment]
> Nice test: https://github.com/cuviper/rust-libprobe/blob/c889fce367aabc78fcd7f347eea7fff569fb6268/examples/loop.rs

I see this as bad formatting; the same thinks rustfmt: https://is.gd/Qh5FCz -> click `Format`

**Comment by jajakobyly** [comment]
> Also, was there #[macro_use] mod foo; as well? 

ups

File: src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs (hunk 1 above)
Comment (jajakobyly) [inline src/test/resources/org/rust/ide/formatter/fixtures/macro_use_after.rs]: Ok I set inlining `#[macro_use]` as the default.

**Comment by matklad** [comment]
Thanks!

**Pull request merged**